### PR TITLE
Add MaxWaitTime

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs [hunk 1]
```diff
@@ -168,19 +168,22 @@ private void CloseLink(ReceivingAmqpLink link)
...
         ///
-        /// <param name="maximumMessageCount">The maximum number of messages that will be received.</param>
+        /// <param name="maxMessages">The maximum number of messages that will be received.</param>
+        /// <param name="maxWaitTime">An optional <see cref="TimeSpan"/> specifying the maximum time to wait for the first message before returning an empty list if no messages have been received.</param>
         /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
         ///
         /// <returns>List of messages received. Returns null if no message is found.</returns>
         public override async Task<IList<ServiceBusReceivedMessage>> ReceiveBatchAsync(
-            int maximumMessageCount,
+            int maxMessages,
```
Comment (ShivangiReja) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs:177]: maxMessages -> maxMessageCount ?

**Review by ShivangiReja — APPROVED** [review]
LGTM

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs:177]: We use maxMessages in the public API.

Comment (ShivangiReja) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs:177]: yeah, I meant can we rename the public API to use maxMessageCount?

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs:177]: I'm not sure the Count adds much as maxMessages seems pretty clear IMO.

Comment (ShivangiReja) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs:177]: I'm fine either way but to me maxMessageCount seems more suitable in this context.

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs:177]: I don't have a strong preference on this either, but I assumed this was out of scope for this PR. I only updated this parameter name for the internal method to be consistent with the other uses. I'd say we can look to see if there are guidelines or examples in other libraries where they append Count, but we can do this for a later preview.

**Pull request merged**